lib/buffer.js

```diff
@@ -658,6 +658,87 @@ Buffer.prototype.includes = function includes(val, byteOffset, encoding) {
   return this.indexOf(val, byteOffset, encoding) !== -1;
 };
 
+Buffer.prototype.indexes = function indexes(match, limit) {
+  if (!Buffer.isBuffer(match))
+    throw new TypeError('argument must be a Buffer');
+  const indexes = [];
+  let i = -1;
+
+  let len = 0
+  while ((i = this.indexOf(match, i + 1)) != -1) {
+    indexes.push(i);
+    len++;
+    if (len >= limit)
+      break;
+  }
+  return indexes;
+};
+
+Buffer.prototype.split = function split(sep, encoding, limit) {
+  if (!limit && typeof encoding === 'number')
+    limit = encoding;
+  if (sep)
+    sep = Buffer.from(sep, encoding); // allows all constructor APIs
+
+  const res = [];
+  // false case: return Array of 1 Buffer
+  const sepLen = sep ? sep.byteLength : undefined;
+  const indexes = sep ? this.indexes(sep, limit) : [];
+
+  if (sepLen > 0) {
+    let cursor = 0;
+    const len = indexes.length;
+    // push from index to index; catch last case after loop
+    for (let i = 0; i < len; i++) {
+      res.push(this.slice(cursor, indexes[i]));
+      cursor += indexes[i] - cursor + sepLen;
+    }
+    // push to the end when no limit is given or when cursor is still below it
+    if (!limit || cursor <= limit)
+      res.push(this.slice(cursor));
+
+  } else if (sepLen === 0) {
+    const len = this.byteLength;
+    // slice through from start to end, byte by byte
+    for (let i = 0; i < len; i++) {
+      res.push(this.slice(i, i + 1));
+    }
+
+  } else if (sepLen === undefined) {
+    // compliance to String.prototype.split; return array of one buffer
+    res.push(this);
+  }
+
+  return res;
+};
+
+// not overriding prototype here, because of Uint8Array, REVIEW
+Buffer.join = function join(array, insert) {
```

Comment:
We happily override Uint8Array methods wherever is convenient. Just make sure it's a superset of functionality, so the Uint8Array method still executes as expected. This should be on the prototype.